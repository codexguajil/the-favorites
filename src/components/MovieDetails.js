import React, { Component } from 'react'

const MovieDetails = ({ title, rating, synopsis, releaseDate, posterImage }) => {
  return (
    <div>
      {title}
      <img src={posterImage} alt='movie poster'/>
      {rating}
      {releaseDate}
      {synopsis}
    </div>
  )
}

export default MovieDetails
