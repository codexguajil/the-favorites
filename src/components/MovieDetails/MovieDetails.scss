.movie-details {
  display: flex;
  width: 75%;
  margin: auto;
  border-radius: 5px;
  background: linear-gradient( rgba(0, 0, 0, 0.6), rgba(0, 0, 0, 0.6))
}

.movie-info {
  color: #fff;
  font-size: 20px;
  display: flex;
  flex-direction: column;
}