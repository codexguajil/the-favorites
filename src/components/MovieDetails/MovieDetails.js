import React, {Component} from 'react';
import './MovieDetails.scss'
import { Link } from 'react-router-dom';

class MovieDetails extends Component {

  render() {
    const { title, rating, posterImage, synopsis, releaseDate } = this.props
    console.log(this.props)
    const image = 'https://image.tmdb.org/t/p/w500'+ posterImage
    return (
      <div>
        <div className='movie-details'>
          <img src={image} alt='movie poster'/>
          <div className='movie-info'>
            <Link to={`/`}><button className='back-btn-fave'>Back To All</button></Link>
            <h2 className='sub-header'>{title}</h2>
            <h3>Rating: {rating}</h3>
            <h3>Release Date: {releaseDate}</h3>
            {synopsis}
          </div>
        </div>
      </div>
    )
  }
}

export default MovieDetails
