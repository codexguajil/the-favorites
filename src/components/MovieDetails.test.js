import React from 'react';
import { shallow } from 'enzyme'
import MovieDetails from './MovieDetails'

describe('MovieDetails', () => {
  it('should match snapshot', () => {
    const wrapper = shallow(
      <MovieDetails />
    )
    expect(wrapper).toMatchSnapshot()
  })
})