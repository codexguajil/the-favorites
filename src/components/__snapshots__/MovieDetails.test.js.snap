// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieDetails should match snapshot 1`] = `
<div>
  <div
    className="movie-details"
  >
    <img
      alt="movie poster"
      src="https://image.tmdb.org/t/p/w500undefined"
    />
    <div
      className="movie-info"
    >
      <Link
        to="/"
      >
        <button
          className="back-btn-fave"
        >
          Back To Movies
        </button>
      </Link>
      <h2
        className="sub-header"
      />
      <h3>
        Rating: 
      </h3>
      <h3>
        Release Date: 
      </h3>
    </div>
  </div>
</div>
`;

exports[`MovieDetails should match snapshot when there is a selected movie 1`] = `
<div>
  <div
    className="movie-details"
  >
    <img
      alt="movie poster"
      src="https://image.tmdb.org/t/p/w500undefined"
    />
    <div
      className="movie-info"
    >
      <Link
        to="/"
      >
        <button
          className="back-btn-fave"
        >
          Back To Movies
        </button>
      </Link>
      <h2
        className="sub-header"
      />
      <h3>
        Rating: 
      </h3>
      <h3>
        Release Date: 
      </h3>
    </div>
  </div>
</div>
`;
